living_documentation_generator/generator.py

```diff
@@ -0,0 +1,316 @@
+
+            if GithubManager().store_repository(repository_id) is None:
+                return {}
+
+            logging.info(f"Downloading issues from repository `{config_repository.owner}/{config_repository.name}`.")
+
+            # Load all issues from one repository (unique for each repository)
+            repo_issues = GithubManager().fetch_issues(labels=config_repository.query_labels)
+            for repo_issue in repo_issues:
+                repo_issue.repository = repository_id
+
+            # save list of repository issue under its repository id
+            # Note: not creating OOP object instead if this dict as Issue type will be replaced in next PR
+            issues[repository_id] = repo_issues
+
+        return issues
+
+    def _mine_github_project_issues(self) -> dict[str, ProjectIssue]:
+        if not self.project_state_mining_enabled:
+            logging.info("Project data mining is not allowed. The process will not start.")
+            return {}
+
+        logging.info("Project data mining allowed, starting the process.")
+
+        # Mine project issues for every repository
+        project_issues = {}
+
+        for config_repository in self.repositories:
+            repository_id = f"{config_repository.owner}/{config_repository.name}"
+
+            if GithubManager().store_repository(repository_id) is None:
+                return {}
+
+            # Fetch all projects_buffer attached to the repository
+            projects = GithubManager().fetch_repository_projects(self.projects_title_filter)
+
+            # Update every project with project issue related data
+            for project in projects:
+                for prj_issue in GithubManager().fetch_project_issues(project):
+                    key = make_issue_key(prj_issue.organization_name, prj_issue.repository_name, prj_issue.number)
+                    project_issues[key] = prj_issue
+
+        return project_issues
+
+    def _consolidate_issues_data(self, gh_repo_issues: dict[str, list[Issue]],
+                                 gh_projects_issues: dict[str, ProjectIssue]) -> dict[str, ConsolidatedIssue]:
+
+        consolidated_issues = {}
+
+        # Create a ConsolidatedIssue object for each repository issue
+        for repository_id in gh_repo_issues.keys():
+            for repo_issue in gh_repo_issues[repository_id]:
+                repo_id_parts = repository_id.split("/")
+                unique_key = make_issue_key(repo_id_parts[0], repo_id_parts[1], repo_issue.number)
+                consolidated_issues[unique_key] = ConsolidatedIssue(repository_id=repository_id, repo_issue=repo_issue)
+
+        # Update issues with project data
+        for key in consolidated_issues.keys():
+            if key in gh_projects_issues:
+                consolidated_issues[key].update_with_project_data(gh_projects_issues[key])
+
+        return consolidated_issues
+
+    def _generate_markdown_pages(self, issues: dict[str, ConsolidatedIssue]):
```

Comment:
Should be static method.

Reply:
Addressed in commit - TBD.

Reply:
Just pointing to the point, that method is not using `self`, so it should have a @staticmethod tag.

Reply:
My bad. Resolving conversation.